SMLHelper/Options/ConfigModOptions.cs

```diff
@@ -0,0 +1,700 @@
+            typeof(OnGameObjectCreatedAttribute), typeof(SliderAttribute), typeof(TooltipAttribute)
+        };
+
+        /// <summary>
+        /// Checks whether a given <see cref="MemberInfo"/> is declared in any subclass of <see cref="ConfigFile"/>.
+        /// </summary>
+        /// <param name="memberInfo">The <see cref="MemberInfo"/> to check.</param>
+        /// <returns>Whether the given <see cref="MemberInfo"/> is declared in any subclass of <see cref="ConfigFile"/>.</returns>
+        private static bool MemberIsDeclaredInConfigFileSubclass(MemberInfo memberInfo)
+            => memberInfo.DeclaringType.IsSubclassOf(typeof(ConfigFile));
+
+        /// <summary>
+        /// Checks whether a given <see cref="MemberInfo"/> should be ignored when generating the options menu, based on whether
+        /// the member has a declared <see cref="IgnoreMemberAttribute"/>, or the <see cref="menuAttribute"/>'s
+        /// <see cref="MenuAttribute.IgnoreUnattributedMembers"/> property.
+        /// </summary>
+        /// <param name="memberInfo">The <see cref="MemberInfo"/> to check.</param>
+        /// <returns>Whether the given <see cref="MemberInfo"/> member should be ignored when generating the options menu.</returns>
+        private bool MemberIsNotIgnored(MemberInfo memberInfo)
+            => !memberInfo.GetCustomAttributes(typeof(IgnoreMemberAttribute), true).Any() &&
+                (!menuAttribute.IgnoreUnattributedMembers || memberInfo.GetCustomAttributes(true)
+                .Any(a => relevantAttributeTypes.Contains(a.GetType())));
+
+        /// <summary>
+        /// Processes the given field or property and hands off to <see cref="AddModOptionMetadata{TModOption}(MemberInfo, Type)"/>
+        /// to generate a <see cref="ModOptionMetadata"/> and add it to the <see cref="modOptionsMetadata"/> dictionary.
+        /// </summary>
+        /// <param name="memberInfo">The <see cref="MemberInfo"/> of the member.</param>
+        /// <param name="memberType">The underlying <see cref="Type"/> of the member.</param>
+        private void ProcessFieldOrProperty(MemberInfo memberInfo, Type memberType)
+        {
+            if (memberType == typeof(bool))
+            {   // Generate a ModToggleOption
+                AddModOptionMetadata<ModToggleOption>(memberInfo, memberType);
+            }
+            else if (memberType == typeof(KeyCode))
+            {   // Generate a ModKeybindOption
+                AddModOptionMetadata<ModKeybindOption>(memberInfo, memberType);
+            }
+            else if (memberType.IsEnum || memberInfo.GetCustomAttributes(typeof(ChoiceAttribute), true).Any())
+            {   // Generate a ModChoiceOption
+                AddModOptionMetadata<ModChoiceOption>(memberInfo, memberType);
+            }
+            else if (new[] { typeof(float), typeof(double), typeof(int) }.Contains(memberType)
```

Comment:
You wouldn't be the first person I see to use `Contains` instead of chaining `||`s.  
Just keep it simple.
```CS
else if (memberType ==  typeof(float) ||
         memberType ==  typeof(double) ||
         memberType ==  typeof(int))
```

Reply:
https://github.com/SubnauticaModding/SMLHelper/pull/173/commits/f1a5ef4766d9b25f06dec66050cd266f6ff9f5ae

Note: after refactor of the attributes (still pending) the Attribute.IsDefined() || chain happening here can potentially be changed to just a single Attribute.IsDefined for the base abstract type. And again, when we get updated to .NET 4.6.2, we can convert all of this to generics.